File: app/src/main/java/com/nutrition/express/common/CommonPagerAdapter.java
package com.nutrition.express.common;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Created by huang on 11/2/16.
 */

public class CommonPagerAdapter extends FragmentPagerAdapter {
    private List<Fragment> list;
    private List<String> titles;

    public CommonPagerAdapter(FragmentManager fm, List<Fragment> list, List<String> titles) {
        super(fm);
        this.list = list;
        this.titles = titles;
    }

    @Override
    public Fragment getItem(int position) {
        return list.get(position);
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles.get(position);
    }

}

File: app/src/main/java/com/nutrition/express/common/CommonPagerAdapter.kt
package com.nutrition.express.common

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

/**
 * Created by huang on 11/2/16.
 */
class CommonPagerAdapter(fm: FragmentManager, private val list: List<Fragment>,
                         private val titles: List<String>) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    override fun getItem(position: Int): Fragment {
        return list[position]
    }

    override fun getCount(): Int {
        return list.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[position]
    }

}